pkg/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,266 @@
+		g := NewWithT(t)
+
+		cli, err := fakeclient.New()
+		g.Expect(err).ShouldNot(HaveOccurred())
+
+		err = upgrade.PatchOdhDashboardConfig(
+			ctx,
+			cli,
+			releaseV1,
+			releaseV2,
+		)
+		g.Expect(err).To(HaveOccurred())
+		g.Expect(err.Error()).To(ContainSubstring("failed to retrieve odhdashboardconfg instance"))
+	})
+
+	t.Run("should return error if updateSpecFields fails", func(t *testing.T) {
+		g := NewWithT(t)
+
+		dashboardConfig := createOdhDashboardConfig()
+		err := unstructured.SetNestedField(dashboardConfig.Object, "invalid_type", "spec")
+		g.Expect(err).ShouldNot(HaveOccurred())
+
+		cli, err := fakeclient.New(dashboardConfig)
+		g.Expect(err).ShouldNot(HaveOccurred())
+
+		err = upgrade.PatchOdhDashboardConfig(
+			ctx,
+			cli,
+			releaseV1,
+			releaseV2,
+		)
+		g.Expect(err).To(HaveOccurred())
+		g.Expect(err.Error()).To(ContainSubstring("failed to update odhdashboardconfig spec fields"))
+	})
+
+	t.Run("should skip patch if no changes are needed", func(t *testing.T) {
+		g := NewWithT(t)
+
+		dashboardConfig := createOdhDashboardConfig()
+		expectedNotebookSizes := []any{
+			map[string]any{"size": "Small", "cpu": "1", "memory": "2Gi"},
+			map[string]any{"size": "Medium", "cpu": "2", "memory": "4Gi"},
+		}
+		expectedModelServerSizes := []any{
+			map[string]any{"size": "Small", "cpu": "2", "memory": "4Gi"},
+		}
+		err := unstructured.SetNestedSlice(dashboardConfig.Object, expectedNotebookSizes, "spec", "notebookSizes")
+		g.Expect(err).ShouldNot(HaveOccurred())
+		err = unstructured.SetNestedSlice(dashboardConfig.Object, expectedModelServerSizes, "spec", "modelServerSizes")
+		g.Expect(err).ShouldNot(HaveOccurred())
+
+		cli, err := fakeclient.New(dashboardConfig)
+		g.Expect(err).ShouldNot(HaveOccurred())
+
+		err = upgrade.PatchOdhDashboardConfig(ctx, cli, releaseV1, releaseV2)
+		g.Expect(err).ShouldNot(HaveOccurred())
+
+		updatedConfig := resources.GvkToUnstructured(gvk.OdhDashboardConfig)
+		err = cli.Get(ctx, client.ObjectKey{Name: dashboardConfig.GetName(), Namespace: dashboardConfig.GetNamespace()}, updatedConfig)
+		g.Expect(err).ShouldNot(HaveOccurred())
+
+		notebookSizes, exists, err := unstructured.NestedSlice(updatedConfig.Object, "spec", "notebookSizes")
+		g.Expect(err).ShouldNot(HaveOccurred())
+		g.Expect(exists).To(BeTrue(), "Expected 'notebookSizes' field to be set")
+		g.Expect(notebookSizes).To(HaveLen(2), "Expected 'notebookSizes' to remain unchanged")
+
+		modelServerSizes, modelServerExists, err := unstructured.NestedSlice(updatedConfig.Object, "spec", "modelServerSizes")
+		g.Expect(err).ShouldNot(HaveOccurred())
+		g.Expect(modelServerExists).To(BeTrue(), "Expected 'modelServerSizes' field to be set")
+		g.Expect(modelServerSizes).To(HaveLen(1), "Expected 'modelServerSizes' to remain unchanged")
+	})
+
+	t.Run("should patch OdhDashboardConfig if changes are needed", func(t *testing.T) {
+		g := NewWithT(t)
+
+		expectedNotebookSizes := []any{
```

Comment:
guess this is the same as `notebookSizesData`  defined in the `upgrade.go` file ?  if so, I think it would be better to reuse it to avoid test and defaults getting out of sync.

Reply:
oops, I did the change in local and forgot to push it. Now its done